arch/xtensa/core/xtensa_mmu.c

```diff
@@ -237,8 +393,8 @@ static void xtensa_mmu_init(bool is_core0)
 	 * Lets use one of the wired entry, so we never have tlb miss for
 	 * the top level table.
 	 */
-	xtensa_dtlb_entry_write(Z_XTENSA_PTE((uint32_t)l1_page_table, MMU_KERNEL_RING,
-				Z_XTENSA_MMU_CACHED_WT),
+	xtensa_dtlb_entry_write(Z_XTENSA_PTE((uint32_t)z_xtensa_kernel_ptables,
+					     MMU_KERNEL_RING, Z_XTENSA_MMU_CACHED_WT),
 			Z_XTENSA_TLB_ENTRY(Z_XTENSA_PAGE_TABLE_VADDR, MMU_PTE_WAY));
 
 	/* Before invalidate the text region in the TLB entry 6, we need to
```

Comment:
This comment is about line 431, but I can't directly comment on that line because it wasn't changed in this review.

With code as-is, I got multi-hit exceptions because way 6 entry 0 and also the kernel page tables mapped to spaces in my SRAM around address 0.

I had to invalidate itlb/dtlb way 6 entry 0 further down in this function, after the vecbase TLB was set up.

Reply:
Yep, that was used to save same pages in the hw that I used to implement it. It is fixed now.